Sources/OAuth1Swift.swift

```diff
@@ -117,9 +117,9 @@ open class OAuth1Swift: OAuthSwift {
     }
 
     @discardableResult
-    open func authorize(withCallbackURL urlString: String, success: @escaping TokenSuccessHandler, failure: FailureHandler?) -> OAuthSwiftRequestHandle? {
-        guard let url = URL(string: urlString) else {
-              failure?(OAuthSwiftError.encodingError(urlString: urlString))
+    open func authorize(withCallbackURL URL: URLConvertible, success: @escaping TokenSuccessHandler, failure: FailureHandler?) -> OAuthSwiftRequestHandle? {
```

Comment:
Could you edit and use `url` instead of `URL` here and for other functions
Thanks